lib/two_bucket_cases.rb

```diff
@@ -0,0 +1,30 @@
+class TwoBucketCase < OpenStruct
+  def test_name
+    "test_bucket_one_#{bucket_one}_bucket_two_"\
+    "#{bucket_two}_goal_#{goal}_starting_#{start_bucket}"
+  end
+
+  def do
+    "TwoBucket.new(#{bucket_one}, #{bucket_two}, #{goal}, '#{start_bucket}')"
+  end
+
+  def test_body
+    "two_bucket = #{self.do}
+    assert_equal #{expected['moves']}, two_bucket.moves
+    assert_equal '#{expected['goal_bucket']}', two_bucket.goal_bucket
+    assert_equal #{expected['other_bucket']}, two_bucket.other_bucket"
+  end
+
+  def skipped?
+    index > 0
+  end
+end
+
+TwoBucketCases = proc do |data|
+  cases = []
+  JSON.parse(data)['two_bucket']['cases'].each_with_index do |row, i|
```

Comment:
I just realized that we could probably use `map.with_index` here instead of `each_with_index`, which would get rid of the `cases` temporary variable.

This is totally not a big deal, and I don't want to hold this back just for this tiny thing.